components/BlogsTable/BlogsTable.tsx

```diff
@@ -0,0 +1,29 @@
+import styles from './blogsTable.module.scss';
+
+type BlogsTableProps<T> = {
+  readonly data: readonly T[];
+  readonly columns: ReadonlyArray<readonly [Partial<keyof T>, string]>;
+};
+
+export function BlogsTable<T>({ data, columns }: BlogsTableProps<T>) {
```

Comment:
Nazwałeś komponent `BlogsTable` i ten komponent jednocześnie jest generyczny. Jeśli to ma być tylko tabela do blogów rozważyłbym pozbycia się generyczności, lub pozbycia się przedrostka `Blogs`.